src/storage/src/source/types.rs

```diff
@@ -39,6 +39,29 @@ use mz_storage_client::types::sources::SourceConnection;
 use crate::source::healthcheck::SourceStatusUpdate;
 use crate::source::metrics::SourceBaseMetrics;
 
+/// Extension trait to the SourceConnection trait that defines how to intantiate a particular
+/// connetion into a reader and offset committer
+pub trait SourceConnectionExt: SourceConnection {
```

Comment:
I think this should be called `SourceReaderBuilder`, not `Ext`; its not really an extension, its a critical trait to implement

Reply:
I was torn about this too. Ideally there would be a single `SourceConnection` trait but since we are splitting the crates for faster compiles there one half in `storage-client` and one here, hence the `Ext` suffix. Neither solutions seem great :/ I changed it to `SourceConnectionBuilder` as you suggest